base_business_document_import/models/business_document_import.py

```diff
@@ -281,9 +281,12 @@ def _match_partner_bank(
             '|', ('company_id', '=', False),
             ('company_id', '=', company_id),
             ('sanitized_acc_number', '=', iban),
+            '|', ('partner_id', '=', False),
             ('partner_id', '=', partner.id),
             ])
         if bankaccounts:
+            if not bankaccounts[0].partner_id:
+                bankaccounts[0].partner_id = partner
```

Comment:
May I suggest to raise an error in this case. This is clearly something fishy with the data and changing it automatically forces us to make assumptions that might be wrong and the user won't be aware of the change.

As the acc_number is unique, we can most likely assume that bankaccounts variable contains 0 or 1 result and we could remove the partner_id part for the search. Because we are in trouble either the partner_id is null or if it is set on an other partner. (This last case could happen in case of duplicates)

We cannot indeed try to create a new bank account in case of bad partner_id, so raising proper exceptions could be a life saver here and let the user figure out how to fixup his data.

Reply:
@alexis-via any opinion on this? What do you prefer?

Better error messages -> instead of a unique number error raise an error to explain the partner setting is wrong on the bank account with IBAN ...
Or
Automatically fixup the partner_id

?